Tests/Common/Data/SliceTests.cs

```diff
@@ -206,6 +207,122 @@ public void AccessesGenericallyByTypeAndSymbol()
+            Slice slice1 = new Slice(DateTime.Today, new BaseData[] { tradeBar1, tick1 });
+            // Use List<tick>
+            var ticks = new Ticks { { Symbols.MSFT, new List<Tick> { tick1 } } };
+            Slice slice3 = new Slice(DateTime.Today, new List<BaseData>(), null, null, ticks, null, null, null, null, null, null);
+            slice1.MergeSlice(slice3);
+            Assert.AreEqual(2, slice1.Ticks.Count);
+        }
+
+        [Test]
+        public void MergeOptionsAndFuturesChain()
+        {
+            // Merge optionChains and FutureChains
+            var optionChain1 = new OptionChains();
+            var optionChain2 = new OptionChains();
+            optionChain1.Add(Symbols.SPY, new OptionChain(Symbols.SPY, DateTime.Now));
+            optionChain2.Add(Symbols.AAPL, new OptionChain(Symbols.SPY, DateTime.Now));
+            var futuresChain1 = new FuturesChains();
+            var futuresChain2 = new FuturesChains();
+            futuresChain1.Add(Symbols.SPY, new FuturesChain(Symbols.SPY, DateTime.Now));
+            futuresChain2.Add(Symbols.AAPL, new FuturesChain(Symbols.SPY, DateTime.Now));
+            var slice4 = new Slice(DateTime.Today, new List<BaseData>(),
+                                new TradeBars(DateTime.Today), new QuoteBars(),
+                                new Ticks(), optionChain1,
+                                futuresChain1, new Splits(),
+                                new Dividends(DateTime.Today), new Delistings(),
+                                new SymbolChangedEvents());
+            var slice5 = new Slice(DateTime.Today, new List<BaseData>(),
+                new TradeBars(DateTime.Today), new QuoteBars(),
+                new Ticks(), optionChain2,
+                futuresChain2, new Splits(),
+                new Dividends(DateTime.Today), new Delistings(),
+                new SymbolChangedEvents());
+            slice4.MergeSlice(slice5);
+            Assert.AreEqual(2, slice4.OptionChains.Count);
+            Assert.AreEqual(2, slice4.FutureChains.Count);
+        }
+
+        [Test]
+        public void MergeCustomData()
+        {
+            TradeBar tradeBar1 = new TradeBar { Symbol = Symbols.SPY, Time = DateTime.Now };
+            TradeBar tradeBar2 = new TradeBar { Symbol = Symbols.AAPL, Time = DateTime.Now, Open = 23 };
+            var custom1 = new FxcmVolume { DataType = MarketDataType.Base, Symbol = Symbols.MSFT };
+            var custom2 = new FxcmVolume { DataType = MarketDataType.Base, Symbol = Symbols.SBIN };
+            Slice slice6 = new Slice(DateTime.Today, new BaseData[] { custom1, custom2, tradeBar2 });
+            Slice slice5 = new Slice(DateTime.Today, new BaseData[] { tradeBar1 });
+            slice5.MergeSlice(slice6);
+            Assert.AreEqual(4, slice5.Values.Count);
+            Assert.AreEqual(2, slice5.Values.Where(x => x.DataType == MarketDataType.Base).Count());
```

Comment:
Cool! similar request as the Ticks tests, let's add custom data with the same symbol in both slices and expect it to use 1 after the merge